services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/GoalStateManagerTest.java

```diff
@@ -0,0 +1,4573 @@
+            + "               \"memoizedHashCode\":-1156439742\n"
+            + "            },\n"
+            + "            \"memoizedIsInitialized\":1,\n"
+            + "            \"unknownFields\":{\n"
+            + "               \"fields\":{\n"
+            + "\n"
+            + "               },\n"
+            + "               \"fieldsDescending\":{\n"
+            + "\n"
+            + "               }\n"
+            + "            },\n"
+            + "            \"memoizedSize\":-1,\n"
+            + "            \"memoizedHashCode\":-1845202229\n"
+            + "         }\n"
+            + "      ],\n"
+            + "      \"securityGroupStates_\":[\n"
+            + "         {\n"
+            + "            \"operationType_\":0,\n"
+            + "            \"configuration_\":{\n"
+            + "               \"bitField0_\":0,\n"
+            + "               \"formatVersion_\":0,\n"
+            + "               \"revisionNumber_\":0,\n"
+            + "               \"id_\":\"\",\n"
+            + "               \"projectId_\":\"\",\n"
+            + "               \"vpcId_\":\"\",\n"
+            + "               \"name_\":\"\",\n"
+            + "               \"securityGroupRules_\":[\n"
+            + "\n"
+            + "               ],\n"
+            + "               \"memoizedIsInitialized\":1,\n"
+            + "               \"unknownFields\":{\n"
+            + "                  \"fields\":{\n"
+            + "\n"
+            + "                  },\n"
+            + "                  \"fieldsDescending\":{\n"
+            + "\n"
+            + "                  }\n"
+            + "               },\n"
+            + "               \"memoizedSize\":-1,\n"
+            + "               \"memoizedHashCode\":0\n"
+            + "            },\n"
+            + "            \"memoizedIsInitialized\":1,\n"
+            + "            \"unknownFields\":{\n"
+            + "               \"fields\":{\n"
+            + "\n"
+            + "               },\n"
+            + "               \"fieldsDescending\":{\n"
+            + "\n"
+            + "               }\n"
+            + "            },\n"
+            + "            \"memoizedSize\":-1,\n"
+            + "            \"memoizedHashCode\":0\n"
+            + "         }\n"
+            + "      ],\n"
+            + "      \"dhcpStates_\":[\n"
+            + "\n"
+            + "      ],\n"
+            + "      \"memoizedIsInitialized\":1,\n"
+            + "      \"unknownFields\":{\n"
+            + "         \"fields\":{\n"
+            + "\n"
+            + "         },\n"
+            + "         \"fieldsDescending\":{\n"
+            + "\n"
+            + "         }\n"
+            + "      },\n"
+            + "      \"memoizedSize\":-1,\n"
+            + "      \"memoizedHashCode\":0\n"
+            + "   }\n"
+            + "}";
+    check(input, output);
+  }
+
+  private void check(String input, String output) {
```

Comment:
The DPM UTs currently cover the GSManager only. We will also need to cover the controller layer and others. 

Please leverage the code coverage report to cover more core logic: https://codecov.io/gh/futurewei-cloud/alcor/tree/f0bcc66f4256ffb75859fc91019bf56bbd68629d/services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane

Create a tracking issue https://github.com/futurewei-cloud/alcor/issues/288.

Reply:
added in 1f5ffa3, would add more